### PR TITLE
[Windows]Fix disk bootorder change after hot-add sata disk

### PR DESCRIPTION
**Pull request by 123lzxm**

### CODE REVIEW

File: windows/vhba_hot_add_remove/set_boot_disk.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#   hdd_name: the specified boot disk name. E.g. 'Hard disk 1' which can be got from vSphere Web UI
+#
+- name: Check required parameter
+  ansible.builtin.assert:
+    that:
+      - hdd_name is defined
+      - hdd_name
+    fail_msg: "hdd_name is not defined"
+
+# Shutdown guest OS
```
Comment (Tomorrow9) [inline windows/vhba_hot_add_remove/set_boot_disk.yml:15]: No need in powered off state to set boot order and boot disk.

File: windows/vhba_hot_add_remove/set_boot_disk.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+#  vars:
+#    set_win_power_state: "shutdown"
+
+# Set the boot disk to the specified disk
+- include_tasks: ../../common/vm_set_boot_options.yml
+  vars:
+    boot_hdd_name: "{{ hdd_name }}"
+    boot_order_list:
+      - disk
+
```
Comment (keirazhang) [inline windows/vhba_hot_add_remove/set_boot_disk.yml:26]: If there is no more includes of `set_boot_disk.yml`, I think this file can be removed, and you can directly put the `include_tasks` into `windows/vhba_hot_add_remove/create_new_file_read_write.yml`

File: windows/vhba_hot_add_remove/set_boot_disk.yml (hunk 1 above)
Comment (123lzxm) [reply windows/vhba_hot_add_remove/set_boot_disk.yml:15]: Thanks.

File: windows/vhba_hot_add_remove/set_boot_disk.yml (hunk 2 above)
Comment (123lzxm) [reply windows/vhba_hot_add_remove/set_boot_disk.yml:26]: Waiting for all the testing results. If no power operations are needed indeed. Will restruct the code.

File: windows/vhba_hot_add_remove/create_new_file_read_write.yml [hunk 3]
```diff
@@ -38,6 +38,27 @@
...
+  block:
+    - name: "Boot order change of hot-add disk to new SATA controller for OS with BIOS firmware"
+      ansible.builtin.debug:
+        msg:
+            - "The disk boot order may be changed after hot-add new disk which causes failure of powering on VM."
+            - "Specify the boot disk as workaround. "
+
+    - include_tasks: ../../common/vm_set_boot_options.yml
+      vars:
+        boot_hdd_name: "Hard disk 1"
```
Comment (Tomorrow9) [inline windows/vhba_hot_add_remove/create_new_file_read_write.yml:51]: Another concern is we need to get VM boot disk name firstly before hotadd testing, then this 'Hard disk 1' is not hardcoded.

Comment (123lzxm) [reply windows/vhba_hot_add_remove/create_new_file_read_write.yml]: Add the logic to get the boot disk name before vhba test.

File: windows/vhba_hot_add_remove/vhba_test_prepare.yml [hunk 4]
```diff
@@ -22,3 +22,37 @@
...
+- name: "Set boot disk label"
+  block:
+    - name: "Get VM VirtualDisk info"
+      include_tasks: ../../common/vm_get_device_with_type.yml
+      vars:
+        device_vim_type: vim.vm.device.VirtualDisk
+
+    - name: "Set the disk label list"
+      ansible.builtin.set_fact:
+        boot_disk_label_list: "{{ boot_disk_label_list|default([]) + [item.deviceInfo.label] }}"
```
Comment (Tomorrow9) [inline windows/vhba_hot_add_remove/vhba_test_prepare.yml:36]: There should be spaces before and after `|`.

File: windows/vhba_hot_add_remove/vhba_test_prepare.yml [hunk 5]
```diff
@@ -22,3 +22,37 @@
...
+    - name: "Display the boot hard disk label"
+      ansible.builtin.debug: var=boot_disk_label
+      when: boot_disk_label is defined
+  when:
+    - test_purpose == "hot-add"
+    - test_disk_controller_type == 'sata'
+    - (vm_firmware | lower == "bios")
+    - win_boot_disk_ctl_type is defined
+    - (win_boot_disk_ctl_type | lower not in ['ide', 'sata'])
+    - add_new_controller
```
Comment (Tomorrow9) [inline windows/vhba_hot_add_remove/vhba_test_prepare.yml:58]: In `vhba_test.yml ` there already has 'add_new_controller' condition.

File: windows/vhba_hot_add_remove/vhba_test_prepare.yml (hunk 4 above)
Comment (123lzxm) [reply windows/vhba_hot_add_remove/vhba_test_prepare.yml:36]: Fixed. Thanks.

File: windows/vhba_hot_add_remove/vhba_test_prepare.yml (hunk 5 above)
Comment (123lzxm) [reply windows/vhba_hot_add_remove/vhba_test_prepare.yml:58]: It's for the second case. I.e add disk to existing controller.

Comment (123lzxm) [reply windows/vhba_hot_add_remove/vhba_test_prepare.yml:58]: Sorry, I misunderstand it. Yes, we can remove it here.

File: windows/vhba_hot_add_remove/create_new_file_read_write.yml [hunk 6]
```diff
@@ -38,6 +38,24 @@
...
+        msg:
+            - "The disk boot order may be changed after hot-add new disk which causes failure of powering on VM."
+            - "Specify the boot disk as workaround. "
+
+    - include_tasks: ../../common/vm_set_boot_options.yml
+      vars:
+        boot_hdd_name: "{{ boot_disk_label }}"
+        boot_order_list:
+          - disk
+  when:
```
[the remark this answers is not in the document]
Comment (123lzxm) [reply windows/vhba_hot_add_remove/create_new_file_read_write.yml:54]: Initialize variable boot_disk_label to empty string in vhba_test_prepare.yml to get the equavelent funcation as you said. Thanks.

**Review by Tomorrow9 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**